cmd/kubermatic-installer/cmd_quickstart.go

```diff
@@ -0,0 +1,377 @@
+	}
+	return nil
+}
+
+func ensureResource(kubeClient client.Client, logger *logrus.Logger, o client.Object) {
+	if err := kubeClient.Create(context.Background(), o); err != nil && !errors.IsAlreadyExists(err) {
+		logger.Fatal(err)
+	}
+}
+
+func installKubevirt(logger *logrus.Logger, dir string, kubeClient client.Client) {
+	logger.Info("Installing KubeVirt ...")
+	// TODO: package this whole thing in KKP charts
+	ko := "https://github.com/kubevirt/kubevirt/releases/download/v1.0.0-alpha.0/kubevirt-operator.yaml"
+	kcr := "https://github.com/kubevirt/kubevirt/releases/download/v1.0.0-alpha.0/kubevirt-cr.yaml"
+	cdio := "https://github.com/kubevirt/containerized-data-importer/releases/download/v1.56.0/cdi-operator.yaml"
+	cdicr := "https://github.com/kubevirt/containerized-data-importer/releases/download/v1.56.0/cdi-operator.yaml"
+	out, err := exec.Command("kubectl", "apply", "-f", ko).CombinedOutput()
+	if err != nil {
+		logger.Fatalf("Failed to install KubeVirt: %v\n%v", err, string(out))
+	}
+	out, err = exec.Command("kubectl", "apply", "-f", kcr).CombinedOutput()
+	if err != nil {
+		logger.Fatalf("Failed to install KubeVirt CR: %v\n%v", err, string(out))
+	}
+	out, err = exec.Command("kubectl", "apply", "-f", cdio).CombinedOutput()
+	if err != nil {
+		logger.Fatalf("Failed to install KubeVirt CDI: %v\n%v", err, string(out))
+	}
+	out, err = exec.Command("kubectl", "apply", "-f", cdicr).CombinedOutput()
+	if err != nil {
+		logger.Fatalf("Failed to install KubeVirt CDI: %v\n%v", err, string(out))
+	}
+}
+
+func installKubermatic(logger *logrus.Logger, dir string, kubeClient client.Client) string {
+	ip := getLocalIP(logger)
+	kkpEndpoint := fmt.Sprintf(kkpEndpointTemplate, ip)
+	logger.Infof("Installing KKP at %v ...", ip) // TODO: prettify
+	valuesExamplePath := filepath.Join(dir, "values.example.yaml")
+	kubermaticExamplePath := filepath.Join(dir, "kubermatic.example.yaml")
+	valuesPath := filepath.Join(dir, "values.yaml")
+	kubermaticPath := filepath.Join(dir, "kubermatic.yaml")
+
+	_, uk, err := loadKubermaticConfiguration(kubermaticExamplePath)
+	if err != nil {
+		logger.Fatalf("failed to unmarshal example kubermatic.yaml: %v", err)
+	}
+	// TODO: make this idempotent (e.g. if generated yamls already exist, don't touch them)
+	unstructured.SetNestedField(uk.Object, kkpEndpoint, "spec", "ingress", "domain")
+	unstructured.SetNestedField(uk.Object, nil, "spec", "ingress", "certificateIssuer")
+	unstructured.SetNestedField(uk.Object, fmt.Sprintf("http://%v/dex", kkpEndpoint), "spec", "auth", "tokenIssuer")
+	unstructured.SetNestedField(uk.Object, randomString(32), "spec", "auth", "issuerClientSecret")
+	unstructured.SetNestedField(uk.Object, randomString(32), "spec", "auth", "issuerCookieKey")
+	unstructured.SetNestedField(uk.Object, randomString(32), "spec", "auth", "serviceAccountKey")
+	// TODO: backport https://github.com/kubermatic/dashboard/pull/5867 or wait for new release?
```

Comment:
TODO: https://github.com/kubermatic/dashboard/pull/5867 is a small change allowing dashboard usage via HTTP (which is my intended way for it to work for quickstart). I will backport it and not override it with a custom dashboard image.

Reply:
it's backported, will wait for KKP v2.22.3 release and then can remove this custom image